sklearn/tree/_splitter.pyx

```diff
@@ -295,8 +318,51 @@ cdef class BestSplitter(BaseDenseSplitter):
                                self.random_state,
                                self.presort), self.__getstate__())
 
+
+    cdef void _shortcut_catlist(self, SIZE_t start, SIZE_t end, INT32_t ncat,
+                                INT32_t ncat_present, const INT32_t *cat_offs,
+                                SIZE_t *shortcut_cat) nogil:
+        """The Breiman shortcut for finding the best split involves a
+        preprocessing step wherein we sort the categories by
+        increasing (weighted) mean of the outcome y (whether 0/1
+        binary for classification or quantitative for
+        regression). This function implements this preprocessing step
+        and produces a sorted list of category values.
+
+        """
+        cdef SIZE_t *samples = self.samples
+        cdef DTYPE_t *Xf = self.feature_values
+        cdef DOUBLE_t *y = self.y
+        cdef SIZE_t y_stride = self.y_stride
+        cdef DOUBLE_t *sample_weight = self.sample_weight
+        cdef DOUBLE_t w
+        cdef SIZE_t cat, localcat
+        cdef SIZE_t q, partition_end
+        cdef DTYPE_t sort_value[64]
+        cdef DTYPE_t sort_den[64]
```

Comment:
I've always wondered if it is cleaner to use
 
```python
cdef:
    SIZE_t *samples = self.samples
    DOUBLE_t sample_weight = self.sample_weight
    ...
```

Would like to know your views on that? :)